navit/traffic/traff_file/traffic_traff_file.c

```diff
@@ -0,0 +1,177 @@
+/**
+ * Navit, a modular navigation system.
+ * Copyright (C) 2005-2017 Navit Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA.
+ */
+
+/**
+ * @file traffic_traff_file.c
+ *
+ * @brief A traff_file traffic plugin.
+ *
+ * This is a traff_file plugin to read traff messages from a file.
+ */
+
+#include <string.h>
+#include <time.h>
+
+#ifdef _POSIX_C_SOURCE
+#include <sys/types.h>
+#endif
+#include "glib_slice.h"
+#include "config.h"
+#include "coord.h"
+#include "item.h"
+#include "xmlconfig.h"
+#include "traffic.h"
+#include "plugin.h"
+#include "debug.h"
+#include <stdlib.h>
+
+long xml_last_msg = 0;
+FILE* traff_file_ptr = 0;
+
+char *traffic_read_traff_file(char *filename);
```

Comment:
Would it make sense to make a header?

Reply:
Other plugins here doesn't have one, so I declared the function in the same way.

Reply:
Okay, I'm fine with that